src/array/cpu/spmm_blocking_libxsmm.h

```diff
@@ -457,10 +457,11 @@ void SpMMRedopCsrOpt(
   const int total_nnz = indptr[M];
   if (M <= 0 || K <= 0 || N <= 0 || total_nnz <= 0) return;
 
-  const float avg_degree = total_nnz * 1.0 / M;
-  const float nnz_prob = avg_degree / K;
+  const double avg_degree = total_nnz * 1.0 / M;
+  const double nnz_prob = avg_degree / K;
```

Comment:
Why does using double fix it?

Reply:
The bug was that on line 463, llc_size/(N * ... ) was coming out to be more than 2^31 for the specific dataset used in #3312. As a result, it was treated as a negative number. Since we are doing a min there, K_block_size was set to a negative number. 
Using double increases the precision of that computation and then typecasting the result to int64 ensures that we are still using high precision.